### DISCUSSION TITLE
Property "$t" was accessed during render but is not defined on instance. @quasar/app-vite and vue-i18n with Legacy API

### DISCUSSION THREAD

**Discussion by lucasdavidferrero** [quasarframework/quasar#12921] [General - Components / Directives / etc] [answered]
This project was created with `quasar create` enabling i18n, Typescript and Vite. I think this error somehow is related with the initialization of vue-i18n.

Error:
```shell
Property "$t" was accessed during render but is not defined on instance.
```

*src/boot/i18n.ts*
```ts
import { boot } from 'quasar/wrappers'
import { createI18n } from 'vue-i18n'

import messages from 'src/i18n/index'

export default boot(({ app }) => {
  const i18n = createI18n({
    locale: 'es',
    fallbackLocale: 'en-US',
    messages
  })

  // Set i18n instance on app
  app.use(i18n)
})
```

*src/i18n/index.ts*
```ts
import enUS from './en-US'
import es from './es'

export default {
  'en-US': enUS,
  es
}
```

*src/i18n/en-US/index.ts*
```ts
export default {
  message: {
    failed: 'Action failed',
    success: 'Action was successful'
  },
  component: {
    familiaProductoSelect: {
      label: 'Family'
    },
    pedidoItem: {
      quantity: 'Quantity',
      removeButton: 'Remove',
      dateAdded: 'Added'
    }
  },
  page: {
    pedidos: {
      mainTitle: 'Your basket ({quantity})',
      noItems: 'No items for now.'
    }
  }
}
```

*src/i18n/es/index.ts* as the same object structure that *en-US*, but in Spanish.

*quasar.config.js*
```js
...
boot: [
      'i18n'
]
```

Using $t in a component:
```vue
<template>
  <div>
    <q-select
      v-model="familiaModel"
      :options="familiaProductos"
      option-value="codigo"
      :option-label="getEntityKeyName"
      emit-value
      map-options
      :label="$t('component.familiaProductoSelect.label')" <!-- This line throws the error.   -->
      outlined
      :loading="getObteniendoInfoApp || getObteniendoListadoProductos"
      :disable="getObteniendoInfoApp || getObteniendoListadoProductos"
      @update:model-value="handleModelUpdate"/>
  </div>
</template>
```
Versions:
"@quasar/app-vite": "^1.0.0-beta.8",
"vue-i18n": "^9.0.0",
"quasar": "^2.6.0",
"@quasar/extras": "^1.0.0",

Anyone knows why is this happening? It was working just fine with webpack.

**Comment by yusufkandemir** [comment, marked as answer]
You need to set [`compositionOnly`](https://github.com/intlify/bundle-tools/tree/vite-plugin-vue-i18n%403.3.1/packages/vite-plugin-vue-i18n#%EF%B8%8F-notice) option of `@intlify/vite-plugin-vue-i18n` to `false` in `quasar.config.js`:
https://github.com/quasarframework/quasar/blob/cebddb615f08e2c32fe2983e4aaaee93744fb93a/create-quasar/templates/app/quasar-v2/ts-vite/BASE/quasar.config.js#L85-L91

**Reply by lucasdavidferrero** [reply]
It works, thanks. Although, I wasn't aware that vue-i18n changed to Composition API, knowing both is useful.

**Reply by CurryAyam** [reply]
Hi,
i'm getting the following error..
`Uncaught (in promise) TypeError: _ctx.$t is not a function`
i've already have the compositionOnly to false

**Reply by 1014670860** [reply]
me too @CurryAyam

**Comment by projik** [comment]
I had the same problem and error message. I was able to fix it by making a change to the boot/i18n.js file.


```
export default boot(({ app }) => {
  const i18n = createI18n({
    locale: "en-US",
    globalInjection: true, // add this line to use $t globally in your app
    messages,
  });

  app.use(i18n);
});
```

I got this fix from this issue on the vue-i18n-next module repo: https://github.com/intlify/vue-i18n-next/issues/350
